keras_retinanet/preprocessing/csv_generator.py

```diff
@@ -31,40 +78,28 @@ def __init__(
         *args,
         **kwargs
     ):
-        self.csv_data_file        = csv_data_file
-
-        self.image_names          = []
-        self.image_data           = {}
+        self.image_names = []
+        self.image_data  = {}
 
         # parse the provided class file
-        self.classes = {}
-        with open(csv_class_file, 'r') as f_class_in:
```

Comment:
Shoud be `'rb'` for windows users.

Reply:
Updated, thanks.

Reply:
But 'r' for Unix ?

Reply:
On unix systems 'b' doesn't do anything.

Reply:
Seems that the documentation is different for python2/3. In the [documentation](https://docs.python.org/2/library/csv.html#csv.reader) for python2 they mention:
```
If csvfile is a file object, it must be opened with the ‘b’ flag on platforms where that makes a difference.
```
Whereas in the python3 [documentation](https://docs.python.org/3/library/csv.html#csv.reader) they say:
```
If csvfile is a file object, it should be opened with newline=''.
```

In addition, a small test on python3, testing `'r'` vs `'rb'` shows the latter doesn't work, with the following error:

```
Error: iterator should return strings, not bytes (did you open the file in text mode?)
```

Reply:
I think this is fixed now. Could you see what you think of it?

Reply:
Should the python3 version not still be opened with `'r'`? Also maybe add a comment for why this is necessary. Other than that, LGTM (unfortunately though).

Reply:
'r' is the default mode for opening file: https://docs.python.org/3/library/functions.html#open

Reply:
Maybe add it anyway for clarity? And comments on why that function is there would be helpful too.

Reply:
I added a docstring and the mode.

Reply:
Then it LGTM. @yhenon ?

Reply:
I think the way suggested here is clearer actually:  https://stackoverflow.com/questions/5180555/python-2-and-3-csv-reader